### PR TITLE
feat: Add the create list modal

### LINKED ISSUE #1715: Create the modal for creating a new list
The form should let users complete the name, description (optional), and if the list is public or private. If any of the required inputs is not completed, the user should not be able to submit the form.

Besides the required inputs, there are 2 important validations to take into account:
1. The name cannot have more than 32 characters
2. The user cannot have more than 1 list with the same name.

Upon creation, the user should be redirected to the new list detail page.

If there is an error in the creation process... TBD.

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the create list modal.
Closes #1715.

Screenshot 2023-05-24 at 16 34 11
Screenshot 2023-05-24 at 16 24 07

### CODE REVIEW

File: webapp/src/modules/favorites/selectors.ts [hunk 1]
```diff
@@ -21,6 +22,9 @@ export const getError = (state: RootState) => getState(state).error
 export const isLoadingFavoritedItems = (state: RootState): boolean =>
   isLoadingType(getLoading(state), FETCH_FAVORITED_ITEMS_REQUEST)
 
+export const isLoadingCreateList = (state: RootState): boolean =>
```
Comment (kevinszuchet) [inline webapp/src/modules/favorites/selectors.ts:25]: Would you mind adding a test for this?

File: webapp/src/components/Modals/CreateListModal/CreateListModal.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  return (
+    <Modal
+      size="tiny"
+      className={styles.modal}
+      onClose={!isLoading ? onClose : undefined}
+    >
+      <ModalNavigation
+        title={t('create_list_modal.title')}
+        onClose={!isLoading ? onClose : undefined}
```
Comment (kevinszuchet) [inline webapp/src/components/Modals/CreateListModal/CreateListModal.tsx:82]: WDYT about moving this to a function called `handleClose`?

File: webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    const { getByTestId } = renderedModal
+    expect(
+      getByTestId(CREATE_LIST_PRIVATE_DATA_TEST_ID).children[0]
+    ).toHaveAttribute('disabled')
+  })
+
+  it('should render the accept button as disabled and loading', () => {
+    const { getByTestId } = renderedModal
+    expect(getByTestId(CREATE_LIST_ACCEPT_BUTTON_DATA_TEST_ID)).toHaveAttribute(
+      'disabled'
```
Comment (kevinszuchet) [inline webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx:68]: You could store the result of `getByTestId(CREATE_LIST_ACCEPT_BUTTON_DATA_TEST_ID)` in a variable to use it in both `expect`. WDYT?

File: webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  })
+
+  it('should render the name input as not disabled', () => {
+    const { getByTestId } = renderedModal
+    expect(
+      getByTestId(CREATE_LIST_NAME_DATA_TEST_ID).children[0]
+    ).not.toHaveAttribute('disabled')
+  })
+
+  it('should render the name input as not disabled', () => {
```
Comment (kevinszuchet) [inline webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx:104]: Would you change this to assert that the description text area is not disabled?

File: webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  })
+
+  it('should render the name input as disabled', () => {
+    const { getByTestId } = renderedModal
+    expect(
+      getByTestId(CREATE_LIST_NAME_DATA_TEST_ID).children[0]
+    ).toHaveAttribute('disabled')
+  })
+
+  it('should render the name input as disabled', () => {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx]: Copy & paste issue 🤦

File: webapp/src/modules/favorites/selectors.ts (hunk 1 above)
Comment (LautaroPetaccio) [reply webapp/src/modules/favorites/selectors.ts:25]: Done!

File: webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx (hunk 3 above)
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx]: Done!

File: webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx (hunk 4 above)
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/CreateListModal/CreateList.spec.tsx]: Fixed it!

**Review by kevinszuchet — APPROVED** [review]

**Pull request merged**